mesecons_luacontroller/init.lua

```diff
@@ -459,6 +459,16 @@ local function get_digiline_send(pos, itbl, send_warning)
 	end
 end
 
+mesecon.luacontroller_libraries = {}
+
+local function get_require(env)
+	return function(name)
+		if mesecon.luacontroller_libraries[name] then
+			return mesecon.tablecopy_change_env(mesecon.luacontroller_libraries[name],env)
+		end
+	end
+end
```

Comment:
You could make this more general by having library getter functions in `mesecon.luacontroller_libraries`:
```suggestion
local function get_require(env, pos)
	return function(name)
		if mesecon.luacontroller_libraries[name] then
			return mesecon.luacontroller_libraries[name](env, vector.new(pos))
		end
	end
end
```

Note that I've also added `pos` here. I think there was somewhen the request to be able to add some sort upgrades to the luacontroller, like a sensor node that is placed beneath the luacontroller, having the pos allows this.

Such a getter function could still change the env of their functions if it wants to. And it has to ensure there are no security flaws, but that's the case anyway, a simpler api without the deepcopy-with-env-change might make this even easier.